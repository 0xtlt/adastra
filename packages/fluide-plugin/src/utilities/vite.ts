// `version` is only exported in Vite 3
import * as vite from 'vite'

export const isViteThree = vite.version?.startsWith('3.')
