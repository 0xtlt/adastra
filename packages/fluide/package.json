{
  "name": "fluide",
  "version": "0.1.0",
  "author": "Blanklob",
  "license": "MIT",
  "type": "module",
  "main": "./dist/index.cjs",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "dev": "npm run build -- --watch",
    "build": "tsup src/index.ts --format cjs,esm --dts",
    "watch": "tsup src/index.ts --format cjs,esm --dts --watch",
    "lint": "eslint",
    "lint:fix": "eslint --fix",
    "lint:types": "tsc --noEmit",
    "test": "vitest run",
    "coverage": "vitest run --coverage"
  },
  "dependencies": {},
  "devDependencies": {
    "@fluide/tsconfig": "workspace:*",
    "eslint-config-fluide": "workspace:*"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/blanklob/fluide.git",
    "directory": "packages/fluide"
  },
  "keywords": [
    "fluide",
    "shopify"
  ]
}
