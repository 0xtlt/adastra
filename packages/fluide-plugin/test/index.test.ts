import path from 'path'
import fs from 'fs/promises'
import { describe, expect, it } from 'vitest'
import { build } from 'vite'

import fluide from '../src'

describe('fluide:build', () => {
  const themeRoot = path.join('test', '__fixtures__')
  it('builds out fluide.liquid snippet file for production', async () => {
    await build({
      logLevel: 'silent',
      plugins: [
        fluide({
          themeRoot,
          sourceCodeDir: path.join(themeRoot, 'src')
        })
      ]
    })

    const tagsLiquid = await fs.readFile(
      path.join(__dirname, '__fixtures__', 'snippets', 'fluide.liquid'),
      { encoding: 'utf8' }
    )
    expect(tagsLiquid).toMatchSnapshot()
  })

  it('builds out fluide.liquid snippet file with hashing', async () => {
    await build({
      logLevel: 'silent',
      plugins: [
        fluide({
          themeRoot,
          sourceCodeDir: path.join(themeRoot, 'src'),
          hash: true,
          minify: false
        })
      ]
    })

    const tagsLiquid = await fs.readFile(
      path.join(__dirname, '__fixtures__', 'snippets', 'fluide.liquid'),
      { encoding: 'utf8' }
    )
    expect(tagsLiquid).toMatchSnapshot()
  })
})
