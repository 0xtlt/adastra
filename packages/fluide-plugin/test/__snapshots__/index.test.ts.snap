// Vitest Snapshot v1

exports[`fluide:build > builds out fluide.liquid snippet file for production 1`] = `
"{% # theme-check-disable %}
{% comment %}
  IMPORTANT: This snippet is automatically generated by Astra.
  Do not attempt to modify this file directly, as any changes will be overwritten by the next build.
{% endcomment %}
{% assign path = fluide | replace: '~/', './' | replace: '@/', './' %}
{% if path == \\"/test/__fixtures__/src/base.css\\" or path == \\"base.css\\" %}
  {{ 'base.css' | asset_url | split: '?' | first | stylesheet_tag: preload: preload }}
{% elsif path == \\"/test/__fixtures__/src/index.ts\\" or path == \\"index.ts\\" %}
  <link rel=\\"modulepreload\\" href=\\"{{ 'index.js' | asset_url | split: '?' | first }}\\" crossorigin=\\"anonymous\\">
  <script src=\\"{{ 'index.js' | asset_url | split: '?' | first }}\\" type=\\"module\\" crossorigin=\\"anonymous\\"></script>
{% endif %}
"
`;

exports[`fluide:build > builds out fluide.liquid snippet file with hashing 1`] = `
"{% # theme-check-disable %}
{% comment %}
  IMPORTANT: This snippet is automatically generated by Astra.
  Do not attempt to modify this file directly, as any changes will be overwritten by the next build.
{% endcomment %}
{% assign path = fluide | replace: '~/', './' | replace: '@/', './' %}
{% if path == \\"/test/__fixtures__/src/base.css\\" or path == \\"base.css\\" %}
  {{ 'base-b66319a3.css' | asset_url | split: '?' | first | stylesheet_tag: preload: preload }}
{% elsif path == \\"/test/__fixtures__/src/index.ts\\" or path == \\"index.ts\\" %}
  <link rel=\\"modulepreload\\" href=\\"{{ 'index-b27101c4.js' | asset_url | split: '?' | first }}\\" crossorigin=\\"anonymous\\">
  <script src=\\"{{ 'index-b27101c4.js' | asset_url | split: '?' | first }}\\" type=\\"module\\" crossorigin=\\"anonymous\\"></script>
{% endif %}
"
`;
